src/openhab.cpp

```diff
@@ -44,29 +44,128 @@ Integration* OpenHABPlugin::createIntegration(const QVariantMap& config, Entitie
         if (iter.key() == "url") {
             _url = iter.value().toString();
-        } else if (iter.key() == "polling_interval") {
-            _pollingInterval = iter.value().toInt();
         }
     }
     if (!_url.endsWith('/')) {
         _url += '/';
     }
+
+    _sseReconnectTimer = new QTimer(this);
+    _sseReconnectTimer->setSingleShot(true);
+    _sseReconnectTimer->setInterval(2000);
+    _sseReconnectTimer->stop();
+
     _pollingTimer.setSingleShot(false);
-    _pollingTimer.setInterval(_pollingInterval);
-    _pollingTimer.stop();
+    _pollingTimer.setInterval(1000 * 60);  // 1 minute
+
+    QObject::connect(&_sseNetworkManager, &QNetworkAccessManager::finished, this, &OpenHAB::streamFinished);
+    QObject::connect(_sseReconnectTimer, &QTimer::timeout, this, &OpenHAB::onSseTimeout);
 
     QObject::connect(&_pollingTimer, &QTimer::timeout, this, &OpenHAB::onPollingTimer);
     QObject::connect(&_nam, &QNetworkAccessManager::networkAccessibleChanged, this, &OpenHAB::onNetWorkAccessible);
 
     qCDebug(m_logCategory) << "setup";
 }
 
+void OpenHAB::streamReceived() {
+    QByteArray     rawData = _sseReply->readAll();
+    QByteArrayList splitted = rawData.split('\n');
+
+    foreach (QByteArray data, splitted) {
+        if (data.length() == 0) continue;
+        if (data.startsWith("event: message")) continue;
+
+        data = data.replace("\"{", "{").replace("}\"", "}").replace("\\\"", "\"");
+
+        QJsonParseError parseerror;
+        // remove the message start passage, because it is no valid JSON
+        QJsonDocument doc = QJsonDocument::fromJson(data.mid(6), &parseerror);
+        if (parseerror.error != QJsonParseError::NoError) {
+            qCCritical(m_logCategory) << "SSE JSON error:" << parseerror.errorString();
+            continue;
+        }
+
+        QVariantMap map = doc.toVariant().toMap();
+        // only process state changes
+        if (!(map.value("topic").toString().endsWith("statechanged"))) continue;
+
+        // get item name from the topic string
+        // example: smarthome/items/EG_Esszimmer_Sonos_CurrentPlayingTime/state
+        QString name = map.value("topic").toString().split('/')[2];
+
+        EntityInterface* entity = m_entities->getEntityInterface(name);
+        if (entity != nullptr) {
+            qCDebug(m_logCategory) << "Debug: " + data;
+
+            QString value = ((QVariantMap*)(map.value("payload").data()))->value("value").toString();
+            // because OpenHab doesn't send the item type in the status update, we have to extract it from our own
+            // entity library
+            if (entity->type() == "light") {
+                processLight(value, entity, false, false);
+            } else if (entity->type() == "blind") {
+                processBlind(value, entity);
+            }
+        } else if (_ohPlayerItems.contains(name)) {
+            QString value = ((QVariantMap*)(map.value("payload").data()))->value("value").toString();
```

Comment:
Could you change the cast to?
`reinterpret_cast<QVariantMap*>(...)`

Reply:
It's also changed.

Reply:
Thank you